src/storage/src/hummock/file_cache/store.rs

```diff
@@ -388,40 +433,33 @@ impl CachedBlock {
         }
     }
 
-    pub fn into_inner(self) -> Box<Block> {
-        match self {
+    pub fn try_into_block(self) -> HummockResult<Box<Block>> {
+        let block = match self {
             CachedBlock::Loaded { block } => block,
-            CachedBlock::Fetched { .. } => unreachable!(),
-        }
+            // for the block was not loaded yet (refill + inflight), we need to decode it.
+            // TODO(MrCroxx): avoid decode twice?
+            CachedBlock::Fetched {
+                bytes,
+                uncompressed_capacity,
+            } => {
+                let block = Block::decode(bytes, uncompressed_capacity)?;
+                Box::new(block)
+            }
```

Comment:
Is this code path reachable? I saw that `CacheBlock::read` will always return `CachedBlock::Loaded`.

Reply:
Yes, it is. When a `CachedBlock::Fetch` is inserted into foyer but not flushed (whose status is also called `Inflight`) and accessed by `lookup`, the code path will be used.

BTW, for the last version of foyer, the code path is unreachable because the inserted entry is serialized into the ring buffer immediately as it is inserted.